.github/workflows/gitversion.yml

```diff
@@ -0,0 +1,49 @@
+# Compute nuget version using GitVersion.
+
+name: GitVersion
+
+on:
+  workflow_call:
+    inputs:
+      prerelease:
+        description: 'Define if this is a prerelease.'
+        required: true
+        type: string
+    outputs:
+      package_version:
+        description: "Computed version for the package (SemVer)"
+        value: ${{ jobs.nuget-version.outputs.version_output }}
+        
+defaults:
+  run:
+    working-directory: src
+    
+jobs:
+  nuget-version:
+
+    runs-on: ubuntu-latest
+    outputs:
+      version_output: ${{ steps.nuget_version.outputs.final_version }}
+    
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+        
+    - name: Git Versioning
+      id: gitversion
+      uses: paulhatch/semantic-version@v4.0.2
+      with:
+        format: "${major}.${minor}.${patch}"
+        
+    # Format is X.X.X-alpha000X.X-X, e.g. 8.1.0-alpha0008.42-1, where 0008 means 8 commits since last tag, 42 is the github run number and 1 is the build try.
+    - name: Append prerelease suffix
+      if: inputs.prerelease == 'true'
+      run: |
+        echo "::set-output name=prerelease_suffix::-alpha$(printf %4s ${{ steps.gitversion.outputs.increment }} | tr ' ' 0).${{ github.run_number }}-${{ github.run_attempt }}"
+      id: prerelease_version
+      
+    - name: Output version
+      run: |
+           echo "::set-output name=final_version::${{ steps.gitversion.outputs.version }}${{ steps.prerelease_version.outputs.prerelease_suffix }}"
+      id: nuget_version
```

Comment:
I get the notification from GitHub that the pipeline have no newline at end of file. Check what that means.

Reply:
This literally just means that there is no new line ate the end. This could confuse some tools in the past. I have added new lines to all .yml files.